lib/knapsack_pro/repository_adapters/git_adapter.rb

```diff
@@ -17,7 +17,13 @@ def branches
       private
 
       def working_dir
-        KnapsackPro::Config::Env.project_dir
+        dir = KnapsackPro::Config::Env.project_dir
+
+        if dir.to_s.include?('~')
+          File.expand_path(dir)
```

Comment:
If the path has `~` sign in `~/repo` then it will be expanded to a full path: `/home/user/repo`.

Thanks to that other parts of the code can work properly, for instance determining git hash and branch name with command like `git -C /home/user/repo rev-parse HEAD` that is called in this class.

Unfortunately command like `git -C ~/repo rev-parse HEAD` crashes because it's not full path.

Reply:
What do you mean by "full path" in the last sentence - do you mean absolute path? Does the problem occur when there is the `~` character in the path specifically, all does it happen with all relative paths? Or maybe it's with all relative paths, but the only ones we encounter contain the tilde?

Basically the reason I'm asking is this: maybe we can always return the absolute path here this way:

```ruby
def working_dir
  dir = KnapsackPro::Config::Env.project_dir

  File.expand_path(dir)
end
```

Reply:
By full path, I meant absolute path. 

Yes, we can do `File.expand_path(dir)` always. I was worried it won't work when someone provides an invalid path (not existing path on the disk) but even then this method works fine so we could use it always.